Library/Homebrew/rubocops/shared/url_helper.rb

```diff
@@ -35,6 +35,12 @@ def audit_urls(urls, regex)
       def audit_url(type, urls, mirrors, livecheck_url: false)
         @type = type
 
+        # URLs must be ASCII; IDNs must be punycode
+        ascii_pattern = /[^\p{ASCII}]+/
+        audit_urls(urls, ascii_pattern) do |_, url|
+          problem "Please use the punycode version of #{url}."
```

Comment:
```suggestion
          problem "Please use the ASCII (punycode, URL encodeded) version of #{url}."
```

Reply:
If we want to be really helpful we would validate the hostname and path separately, and write the correct error message and correctly encoded expected full URL.